it/src/main/java/com/google/cloud/teleport/it/bigtable/BigtableResourceManagerClientHandler.java

```diff
@@ -0,0 +1,100 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.it.bigtable;
+
+import com.google.cloud.bigtable.admin.v2.BigtableInstanceAdminClient;
+import com.google.cloud.bigtable.admin.v2.BigtableInstanceAdminSettings;
+import com.google.cloud.bigtable.admin.v2.BigtableTableAdminClient;
+import com.google.cloud.bigtable.admin.v2.BigtableTableAdminSettings;
+import com.google.cloud.bigtable.data.v2.BigtableDataClient;
+import com.google.cloud.bigtable.data.v2.BigtableDataSettings;
+import java.io.IOException;
+
+/**
+ * Class for storing the 3 Bigtable clients used to manage the Instance(s), Table(s) and Data
+ * mutations performed by the resource manager.
+ *
+ * <p>This class is intended to simplify mock injections of Bigtable clients during unit testing
+ */
+public class BigtableResourceManagerClientHandler {
+
+  private final BigtableInstanceAdminClient bigtableInstanceAdminClient;
+  private final BigtableTableAdminClient bigtableTableAdminClient;
+  private final BigtableDataSettings bigtableDataSettings;
+
+  private BigtableResourceManagerClientHandler(BigtableResourceManagerClientHandler.Builder builder)
+      throws IOException {
+    this.bigtableInstanceAdminClient = BigtableInstanceAdminClient.create(builder.instanceSettings);
+    this.bigtableTableAdminClient = BigtableTableAdminClient.create(builder.tableSettings);
+    this.bigtableDataSettings = builder.dataSettings;
+  }
+
+  public static BigtableResourceManagerClientHandler.Builder builder(
+      BigtableInstanceAdminSettings instanceSettings,
+      BigtableTableAdminSettings tableSettings,
+      BigtableDataSettings dataSettings) {
+    return new BigtableResourceManagerClientHandler.Builder(
+        instanceSettings, tableSettings, dataSettings);
+  }
+
+  /**
+   * Returns the Instance admin client for managing Bigtable instances.
+   *
+   * @return the instance admin client
+   */
+  public BigtableInstanceAdminClient getBigtableInstanceAdminClient() {
+    return bigtableInstanceAdminClient;
+  }
+
+  /**
+   * Returns the Table admin client for managing Bigtable tables.
+   *
+   * @return the table admin client
+   */
+  public BigtableTableAdminClient getBigtableTableAdminClient() {
+    return bigtableTableAdminClient;
+  }
+
+  /**
+   * Returns the Data client for reading and writing data to Bigtable tables.
+   *
+   * @return the data client
+   */
+  public BigtableDataClient getBigtableDataClient() throws IOException {
+    return BigtableDataClient.create(bigtableDataSettings);
+  }
+
+  /** Builder for {@link BigtableResourceManagerClientHandler}. */
+  public static final class Builder {
+
+    private final BigtableInstanceAdminSettings instanceSettings;
+    private final BigtableTableAdminSettings tableSettings;
+    private final BigtableDataSettings dataSettings;
+
+    private Builder(
```

Comment:
Same: no need for a builder if all its properties are required.

Reply:
Fixed.